management/src/clusterm/manager/client.go

```diff
@@ -93,55 +99,67 @@ func (c *Client) doGet(rsrc string) ([]byte, error) {
 }
 
 // PostNodeCommission posts the request to commission a node
-func (c *Client) PostNodeCommission(nodeName, extraVars string) error {
-	return c.doPost(fmt.Sprintf("%s/%s", PostNodeCommissionPrefix, nodeName), extraVars, nil)
+func (c *Client) PostNodeCommission(nodeName string, flags ActionFlags) error {
+	if !IsValidHostGroup(flags.HostGroup) {
```

Comment:
this validation shall happen in the backend (i.e. in the event processor). This ensures that someone calling the REST API directly (like our web UI) will get the same behavior as someone using clusterctl or this go-client.